### PR TITLE
Set title for new chat

### PR DESCRIPTION
**Pull request by thehunmonkgroup**
Uses the `/backend-api/conversation/gen_title/[id] POST endpoint`

Note this PR requires PR #109 to function properly.

One optimization that could be made before committing: the POST request is a sync request after the very first message is sent in a new conversation, and the time it takes delays the display of prompt `2>` until the request completes. Once the title is set there is no delay for other messages in the conversation.

Also note that I abstracted an `_api_post_request()` method, and with a little more work, it would be trivial to add other helpful API requests, such as #106.

### CODE REVIEW

**Comment by thehunmonkgroup** [comment]
Note this will fix #110

File: chatgpt_wrapper/chatgpt.py [hunk 1]
```diff
@@ -102,6 +110,34 @@ def _cleanup_divs(self):
...
+        headers.update(custom_headers)
+        return headers
+
+    def _api_post_request(self, url, data={}, custom_headers={}):
+        headers = self._api_request_build_headers(custom_headers)
+        response = self.page.request.post(url, headers=headers, data=data)
+        return response
+
+    def _set_title(self):
+        if not self.conversation_id or self.conversation_id and self.conversation_title_set:
```
Comment (mmabrouk) [inline chatgpt_wrapper/chatgpt.py:126]: it looks like some parenthesis are missing here

File: chatgpt_wrapper/chatgpt.py [hunk 2]
```diff
@@ -54,7 +60,9 @@ def __init__(self, headless: bool = True, browser="firefox", timeout=60, proxy:
         self._start_browser()
         self.parent_message_id = str(uuid.uuid4())
         self.conversation_id = None
+        self.conversation_title_set = None
```
Comment (mmabrouk) [inline chatgpt_wrapper/chatgpt.py:63]: could you please rename this to clarify it's binary:
```suggestion
        self.is_conversation_title_set = False
```

File: chatgpt_wrapper/chatgpt.py [hunk 3]
```diff
@@ -250,3 +287,4 @@ def ask(self, message: str) -> str:
     def new_conversation(self):
         self.parent_message_id = str(uuid.uuid4())
         self.conversation_id = None
+        self.conversation_title_set = None
```
Comment (mmabrouk) [inline chatgpt_wrapper/chatgpt.py:290]:
```suggestion
        self.is_conversation_title_set = False
```

**Review by mmabrouk — CHANGES_REQUESTED** [review]
Thanks for fixing the issue! Much appreciated.
I had very minor comments for the code.

**Comment by mmabrouk** [comment]
Since I approved the !delete PR which includes this patch, I will close this one and make the minor changed myself on the main branch

**Pull request closed without merging**